examples/server/pkg/lifecycle/lifecycle.go

```diff
@@ -68,7 +68,7 @@ func notifyResourceChanges(s server.Server) {
 // Start starts the server lifecycle.
 func Start(rootCACerts []string, peerCert string, peerKey string, port uint32) {
 	// Initialize the server.
-	tlsConfig := tls.PrepareServerConfig(rootCACerts, peerCert, peerKey)
+	tlsConfig := tls.PrepareMutualTLSServerConfig(rootCACerts, peerCert, peerKey)
```

Comment:
@andrewsykim since the new helper functions are not documented anywhere (apart from the issue, that is going to fade away with time), it may be good if at least you refer here that it exists an alternative helper function. Otherwise, I fear nobody will notice.